### PR TITLE
rename digest to content-digest for clarity

### LINKED ISSUE #159: rename digest field to contentDigest

### PR DESCRIPTION
**Pull request by michelleN**
as suggested by @simonferquel 
resolves #159

### CODE REVIEW

File: 103-bundle-runtime.md [hunk 1]
```diff
@@ -195,13 +195,13 @@ For this example CNAB bundle:
...
       "description": "my microservice",
-      "digest": "sha256:aaaaaaaaaaaa...",
+      "content-digest": "sha256:aaaaaaaaaaaa...",
       "image": "technosophos/microservice:1.2.3"
     }
   },
   "invocationImages": [
     {
-      "digest": "sha256:aaaaaaa...",
+      "content-digest": "sha256:aaaaaaa...",
```
Comment (radu-matei) [inline 103-bundle-runtime.md]: I think there is one more reference of `digest` in this file, in the example below (line 240): "The `/cnab/app/image-map.json` file mounted in the invocation image will be"

Comment (michelleN) [reply 103-bundle-runtime.md]: good catch. updated.

File: 100-CNAB.md [hunk 2]
```diff
@@ -29,7 +29,7 @@ The canonical encoding of a bundle definition is a JSON-formatted file, which MU
 
 In either case, CNAB has the same schema, and this spec refers to this file as the "bundle definition" (or occasionally "bundle file"). 
 
-However, as a signed bundle definition represents an immutable bundle, all invocation images and images references must have a digest.
+However, as a signed bundle definition represents an immutable bundle, all invocation images and images references must have a content-digest.
```
Comment (simonferquel) [inline 100-CNAB.md]: or have a digested image reference (e.g.: myimage@sha256:aabb...)

**Comment by simonferquel** [comment]
I am not sure about the correct wording for my comment above, but I think we really need some clarification here.

**Comment by radu-matei** [comment]
@michelleN, @simonferquel - should I add this topic for this week's CNAB meeting?
I know @michelleN probably won't be able to make it to the meeting, but I think we can at least clarify what we're after here.

File: schema/bundle.schema.json [hunk 3]
```diff
@@ -131,7 +131,7 @@
                     "type": "string",
                     "default": "oci"
                 },
-                "digest":{
+                "content-digest":{
```
Comment (jlegrone) [inline schema/bundle.schema.json]: Seems like there's a strong convention around camel cased keys.

File: 100-CNAB.md (hunk 2 above)
Comment (michelleN) [reply 100-CNAB.md]: I disagree here. At the point at which a bundle is ready to be shared or shipped, a bundle should have a content digest regardless of whether the image reference contains a digest or not. Do you disagree?

**Comment by michelleN** [comment]
@simonferquel Since this comment was from a long while ago, do you still feel thin bundles should not have content digest. I think both thin/thick bundles rather any bundle that is ready to be shared or shipped should contain content digests for each image it references. What are your current thoughts around this?

>It would be nice to clarify the content-digest conditional requirements:

>>if in-dev: no digest required, but bundle is not shareable
>>if thick-bundle: content-digest is required
>>if thin-bundle: content-digest OR digested reference is required

**Comment by michelleN** [comment]
@technosophos @jlegrone updated. thanks

**Comment by simonferquel** [comment]
My point was more about "in-dev" bundles: basically a bundle is not ready to be shared until everything is either referenced by digest or has a contentDigest with it. But for making things easier during development, the runtime should be capable of accepting not-digested reference (e.g. images I just built locally). The spec does not have to specify how the tooling should enforce this though.

**Comment by simonferquel** [comment]
otherwise, LGTM

**Comment by technosophos** [comment]
@michelleN This needs a rebase, but I think you can merge after that (unless you wanted to address Simon's last comment)

**Comment by michelleN** [comment]
@simonferquel - I added some wording that addresses your concerns. Lmk what you think!

**Pull request merged**